### ISSUE TITLE
Add search for nodes in GistFile TreeView

### ISSUE THREAD

**Issue by dustinkredmond** [RedmondSims/GistFX#6]
Need to look at adding a search field within the TreeView to allow searching for content in the GistFiles. A single search field with fuzzy searching on GistFile name would provide a big improvement. Allowing to search on source code within each file may be better, but may also prove to be a more expensive operation due to the speed limitations imposed by the GitHub API.

**Comment by EasyG0ing1** [comment]
Do you know of any good libraries that already exist that can compile words
and associate connections to the objects that contain the words? That would
be handy ... no need to re-invent the wheel if not necessary.

But I did actually implement this once, but then it got shelved for other
things ... but I had forgotten about it, so I'm glad you brought it up.

Should we create issues inside the repository for these changes?

**Comment by dustinkredmond** [comment]
I agree, probably best not to reinvent the wheel here. I will need to do some research. There are some good Python libraries that I know off of the top of my head, but none in Java that I can think of.

**Comment by EasyG0ing1** [comment]
By the way, I'm not entirely sold on the Toolbar icon business ... that was
kind of an experiment ... I use to just have buttons there but I thought it
might be nice to have icons instead but I think in my head it was different
than how it turned out so if you're not a fan of the toolbar, let me know
... I'm not opposed to shit canning it all together.

**Comment by EasyG0ing1** [comment]
Well, If it exists in Python, then it must exist somewhere in Java ....

Then again, I was looking for a program to convert an FXML file into a java
class and I only found TWO - one was in Python and the other in NetBeans
version 8.2 as a plugin ... so I did what any Python resistor would do and
I fired up a virtual machine and found NetBeans 8.2 in an old dusty virtual
basement somewhere and I installed it and I converted some FXML into java
classes ... which is actually the three tabs on the settings screen in
GistFX where you can change the tree icons ... I couldn't get the layout
consistent and I didn't want to mess with GridPane manually because it
never is much of a good time when I try GridPane so I did a Gridpane in
SceneBuilder then made it into a java class with old NetBeans.

By the way, ill be pushing new binaries tonight ... code is much tighter
... settings screen is almost done ... and I added the ability to re-size
the toolbar icons, which I'm sure you'll appreciate.

Did you try the detached Toolbar?  That took some creative coding to get
working properly.

**Comment by EasyG0ing1** [comment]
@dustinkredmond - I added a very basic search feature above the tree. You won't see it, but if you move the mouse pointer to just above the tree, you'll see it turn into an I beam ... 

<img width="309" alt="Screen Shot 2022-04-14 at 6 37 03 PM" src="https://user-images.githubusercontent.com/10106834/163504239-9f8f5fcc-66a9-45aa-903e-9250a0e9e54e.png">


You have to click once in the text field as that primes the Search class with the treeRoot so it can build everything it needs to do the search.

So just click in the search field, start typing, and when you see a number of hits that seem relevant, just keep hitting the enter key and it will cycle through the hits for you.

Download 3.4.43 in releases.

It uses the text from the gist name, description, file name, and file description (if you added any) ... which, by the way, is what the TextArea is above the code window when you have a file selected ... you can type in a description for the file itself... the search searches all text except any file contents ... if you want it to include the contents of the files in the search, just go to Edit / Settings and in the first tab, you can enable that.

Mike

**Comment by EasyG0ing1** [comment]
@dustinkredmond - I added the ability to include file contents ... just go to settings and it's in the first tab.

Download version 3.4.42

**Comment by EasyG0ing1** [comment]
@dustinkredmond 

Download version 3.4.43 - improved .42 version
